raster-foundry/raster-foundry
# Add names to tool runs, refactor tools ui for future work

## Overview

Add name to tool_runs
Add name to tool_runs update fields
Merge metadata updates instead of overwriting
Fix missing input node labels
Add ability to save node position
Remove old components
Fix nvd3 dependency

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Notes
Depends on #2515 
In general, old tool runs are broken in a bunch of ways due to changing ast formats. It's probably work running `delete from tool_runs` just to clean things up.

## Testing Instructions

* Run migrations
* Verify that tool_runs now have names, and have defaulted to the name of the tools they were based on
* Verify that tool_runs no longer reference the tools they are created from
* Verify that you can create a tool from a template
  * I reference tool_runs as tools on the frontend, since we're calling tools templates. Be aware of this to avoid a headache
* Verify that you can't preview tools before they are created - a tool is only created when provided with valid inputs
* Verify that node placement can now be changed, and is saved
* Verify that once a tool is created, it can be previewed immediately
* Verify that saving a tool doesn't delete the node names
* Verify that histograms work again

Closes #2640 
Closes #2628
